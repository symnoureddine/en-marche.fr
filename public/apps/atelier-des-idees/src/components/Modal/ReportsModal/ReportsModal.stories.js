import React from 'react';
import { storiesOf } from '@storybook/react';
import ReportsModal from '.';

const props = {
    reports: [
        {
            file: '/test',
            fileName: 'document-5.pdf',
            size: '1.2 Mb',
        },
        {
            file: '/test',
            fileName: 'document-5.pdf',
            size: '1.2 Mb',
        },
        {
            file: '/test',
            fileName: 'document-5.pdf',
            size: '1.2 Mb',
        },
        {
            file: '/test',
            fileName: 'document-5.pdf',
            size: '1.2 Mb',
        },
        {
            file: '/test',
            fileName: 'document-5.pdf',
            size: '1.2 Mb',
        },
    ],
};

storiesOf('ReportsModal', module)
    .addParameters({ jest: ['ReportsModal'] })
    .add('default', () => <ReportsModal {...props} />);
