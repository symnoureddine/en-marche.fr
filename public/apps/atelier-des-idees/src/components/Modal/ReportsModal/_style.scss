.reports-modal {
	&__subtitle {
		color: $light-black;
		margin-bottom: 52px;
	}

	&__report {
		display: flex;
		width: 500px;
		padding: 12px 0 12px 12px;
		margin-bottom: 5px;
		text-decoration: none;
		border: none;

		&__content {
			display: flex;
			flex-direction: column;
			justify-content: center;
			margin-left: 15px;

			&__file {
				color: $saphir;
			}

			&__size {
				color: $wild-blue-yonder;
				font-size: 14px;
			}
		}
	}

	&__report:hover {
		background: $white;
		border-radius: 5px;
	}

	/* Responsive */
	@media #{$mobile} {
		overflow-x: hidden;
	}
}
