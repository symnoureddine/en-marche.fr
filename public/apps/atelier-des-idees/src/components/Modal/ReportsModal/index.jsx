import React from 'react';
import PropTypes from 'prop-types';
import pdf_preview from './../../../img/pdf_preview.svg';

function ReportsModal(props) {
    return (
        <div className="reports-modal">
            <h2 className="reports-modal__title">Je lis les {props.reports.length} restitutions</h2>
            <p className="reports-modal__subtitle">Nous n’avons pas attendu l’Atelier des idées pour vous consulter</p>
            {props.reports.map((report, index) => (
                <a className="reports-modal__report" key={index} target="_blank" href={report.url} download>
                    <img className="reports-modal__report__pdf-preview" src={pdf_preview} />
                    <div className="reports-modal__report__content">
                        <span className="reports-modal__report__content__file">{report.name}</span>
                        {report.size && <span className="reports-modal__report__content__size">{report.size}</span>}
                    </div>
                </a>
            ))}
        </div>
    );
}

ReportsModal.defaultProps = {
    reports: [],
};

ReportsModal.propTypes = {
    reports: PropTypes.arrayOf(
        PropTypes.shape({
            url: PropTypes.string,
            name: PropTypes.string,
            size: PropTypes.string,
        })
    ),
};

export default ReportsModal;
