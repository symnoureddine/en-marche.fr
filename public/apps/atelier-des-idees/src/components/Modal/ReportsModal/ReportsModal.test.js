import React from 'react';
import TestRenderer from 'react-test-renderer';
import ReportsModal from '.';

describe('ReportsModal', () => {
    it('renders correctly', () => {
        const tree = TestRenderer.create(<ReportsModal />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
