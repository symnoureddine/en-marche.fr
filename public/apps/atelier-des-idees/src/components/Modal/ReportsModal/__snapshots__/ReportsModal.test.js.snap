// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportsModal renders correctly 1`] = `
<div
  className="reports-modal"
>
  RAPPORTS
</div>
`;
